Subscriptions docs1 (#716)
aerogear/mobile-docs: modules/ROOT/pages/_partials/data-sync/realtime-intro.adoc

[id="realtime-intro-{context}"]
= Introduction to real-time updates

After you have developed some queries and mutations, you might want to implement real-time updates.
These are supported in the GraphQL specification by an operation type called `Subscription`.
To support scalable, reliable subscriptions in a production environment, Data Sync implements subscriptions using an MQTT PubSub subscription mechanism, however you might want to use the Apollo PubSub module to develop proof-of-concept applications.  

When coding for real-time updates, you use the following modules:

* @aerogear/voyager-server - supports clients that use voyager-client to enable GraphQL queries and mutations
* @aerogear/voyager-subscriptions - supports clients that use voyager-client to enable GraphQL subscriptions
* @aerogear/graph-mqtt-subscriptions - supports GraphQL resolvers connections to a MQTT broker

> GraphQL Subscriptions utilise websockets to enable clients to subscribe to published changes.

Reviewer: I might suggest something like `GraphQL Subscriptions enable clients to subscribe to events <generated within|that happen on> the server over a websocket connection`

Author: updated

> The architecture of websockets is:

Reviewer: This title is a little inaccurate and possibly confusing. Websockets is a lower level communication protocol, over which subscriptions are delivered.

I would maybe avoid using the term `websocket server` too because it's possible that someone would read that as a different server to the data sync server (which is not the case).

I would maybe change the title here to `The architecture of subscriptions is` or `The general flow for subscriptions is` or something like that.

As for the bullet points themselves, I got a strong impression here that any and all clients receive every single event that happens on the server, which is not the case. As part of the initial connection, the client **Subscribes** to particular events it is interested in. (example in a chat app: I want to know about messages only in a particular channel. Not all messages in the whole universe). I might go with something like this

```
* Client connects to the server using websockets, and subscribes to certain events
* Upon events, the server notifies the clients that are <subscribed to|interested in> those events
* Any _currently connected_ client that is subscribed to a given event receives it.
* The client can close the connection at any time and no longer receives updates.
```

Author: updated

* Client connects to websocket server.
* Upon certain events, the server can publish the results of these events to the websocket.
* Any _currently connected_ client to that websocket receives these results.
* The client can close the connection at any time and no longer receives updates.

> To receive updates the client must be currently connected to the websocket server, updates made over this websocket while the client is offline are not consumed by the client.

Reviewer: ```suggestion
To receive updates, the client must be currently connected to the server, the client will not receive events from subscriptions while offline.
```

Mainly trying to avoid the sentence `updates made over this websocket` which is inaccurate.

Author: updated
